### PR TITLE
Preliminary solution to 1594->create bip service 

### PR DESCRIPTION
**Pull request by engineer-plus-plus**
The preliminary solution to [this story](https://github.com/department-of-veterans-affairs/abd-vro/issues/1594)

## What was the problem?
BIP api is just a bunch of classes that have no way of interacting with it.

Associated tickets or Slack threads:
all associated things are listed in [this story](https://github.com/department-of-veterans-affairs/abd-vro/issues/1594)

## How does this fix it?[^1]
Adds dependencies and annotations that cause it to listen to messages in rabbit mq.  Adds configuration to stand it up in a docker container.

## How to test this PR
Same way as in `svc-lighthouse-api`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

File: svc-lighthouse-api/build.gradle [hunk 1]
```diff
@@ -30,6 +30,6 @@ dependencies {
 }
 
 
-test {
-  useJUnitPlatform()
-}
+//test {
```
Comment (engineer-plus-plus) [inline svc-lighthouse-api/build.gradle:33]: Not to self: this has to be uncommitted.

File: svc-bip-api/build.gradle [hunk 2]
```diff
@@ -1,6 +1,8 @@
 plugins {
-  id 'java'
-  id 'local.std.java.library-conventions'
+  id 'local.java.container-spring-conventions'
+  id 'local.std.java.library-spring-conventions'
+  id 'io.spring.dependency-management'
+  id 'jacoco'
```
Comment (yoomlam) [inline svc-bip-api/build.gradle:5]: You probably don't need these
```suggestion
```

File: domain-rrd/gradle.properties [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (yoomlam) [inline domain-rrd/gradle.properties:1]:
```suggestion
```

File: svc-bip-api/build.gradle [hunk 4]
```diff
@@ -13,4 +15,21 @@ dependencies {
...
+  implementation "com.fasterxml.jackson.core:jackson-databind:${jackson_databind_version}"
+  implementation "com.fasterxml.jackson.core:jackson-core:${jackson_version}"
+  implementation 'org.springframework.boot:spring-boot-starter-amqp'
+  implementation 'org.springframework.boot:spring-boot-starter-web'
+  implementation 'org.springframework.boot:spring-boot-starter-actuator'
+  implementation 'org.springframework:spring-web:5.3.23'
+  implementation 'ca.uhn.hapi.fhir:hapi-fhir-base:6.6.0'
+  implementation('ca.uhn.hapi.fhir:hapi-fhir-server-openapi:6.6.0') {
+    exclude group: 'xalan', module: 'xalan'
+  }
```
Comment (yoomlam) [inline svc-bip-api/build.gradle:28]: I don't think these are needed for BIP
```suggestion
```

File: svc-bip-api/src/docker/docker-compose.yml
Comment (yoomlam) [inline svc-bip-api/src/docker/docker-compose.yml:1]: Delete this file, unless there's a particular reason for it.

File: svc-bip-api/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+## RabbitMQ Settings
+
+spring.rabbitmq.host=${RABBITMQ_PLACEHOLDERS_HOST}
+spring.rabbitmq.port=5672
+spring.rabbitmq.username=${RABBITMQ_PLACEHOLDERS_USERNAME}
+spring.rabbitmq.password=${RABBITMQ_PLACEHOLDERS_USERPASSWORD}
+
+abd-data-access.rabbitmq.claim-submit-exchange=claim-submit-exchange
+abd-data-access.rabbitmq.claim-submit-queue=claim-submit
+abd-data-access.rabbitmq.claim-submit-routing-key=code.#
```
Comment (yoomlam) [inline svc-bip-api/src/main/resources/application.properties:32]: `abd-data-access` was/is used for the `svc-lighthouse-api` client. Use something else like `bip-api-client`.

File: svc-bip-api/src/main/java/gov/va/vro/bip/config/RabbitMqConfig.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+@Configuration
+public class RabbitMqConfig {
+  @Value("${abd-data-access.rabbitmq.claim-submit-exchange}")
+  private String claimSubmitExchangeName;
+
+  @Value("${abd-data-access.rabbitmq.claim-submit-queue}")
+  private String claimSubmitQueueName;
+
+  @Value("${abd-data-access.rabbitmq.claim-submit-routing-key}")
+  private String claimSubmitRoutingKey;
```
Comment (yoomlam) [inline svc-bip-api/src/main/java/gov/va/vro/bip/config/RabbitMqConfig.java:22]: Shouldn't these settings and variable names correspond to the queues mentioned in `BipApiService`?

File: svc-bip-api/src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,49 @@
+# FIXME: Why is this needed?
+server.port=8091
```
Comment (yoomlam) [inline svc-bip-api/src/main/resources/application.properties:2]:
```suggestion
```

File: svc-bip-api/src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+spring.rabbitmq.password=${RABBITMQ_PLACEHOLDERS_USERPASSWORD}
+
+abd-data-access.rabbitmq.claim-submit-exchange=claim-submit-exchange
+abd-data-access.rabbitmq.claim-submit-queue=claim-submit
+abd-data-access.rabbitmq.claim-submit-routing-key=code.#
+
+## Lighthouse FHIR API
+
+lh.clientId=${LH_ACCESS_CLIENT_ID}
+lh.pemkey=${LH_PRIVATE_KEY}
```
Comment (yoomlam) [inline svc-bip-api/src/main/resources/application.properties:37]:
```suggestion
```

**Review by yoomlam — CHANGES_REQUESTED** [review]

**Comment by engineer-plus-plus** [comment]
closing pull request as the final solution will use a different model (svc-xample-j)

**Pull request closed without merging**